core/types/src/svn.rs

```diff
@@ -2,41 +2,73 @@
 //! SGX core SVN (Security Version Numbers)
 
 use crate::{impl_newtype, impl_newtype_for_bytestruct};
+use constant_time_derive::ConstantTimeEq;
 use mc_sgx_core_sys_types::{sgx_config_svn_t, sgx_cpu_svn_t, sgx_isv_svn_t, SGX_CPUSVN_SIZE};
+use subtle::{Choice, ConstantTimeGreater, ConstantTimeLess};
 
 /// Config security version number (SVN)
 #[repr(transparent)]
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Default)]
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Default, ConstantTimeEq)]
 pub struct ConfigSvn(sgx_config_svn_t);
 
 impl_newtype! {
     ConfigSvn, sgx_config_svn_t;
 }
 
+impl ConstantTimeGreater for ConfigSvn {
+    fn ct_gt(&self, other: &Self) -> Choice {
+        self.0.ct_gt(&other.0)
+    }
+}
+
+impl ConstantTimeLess for ConfigSvn {}
+
 /// Independent software vendor (ISV) security version number (SVN)
 #[repr(transparent)]
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Default)]
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Default, ConstantTimeEq)]
 pub struct IsvSvn(sgx_isv_svn_t);
 
 impl_newtype! {
     IsvSvn, sgx_isv_svn_t;
 }
 
+impl ConstantTimeGreater for IsvSvn {
+    fn ct_gt(&self, other: &Self) -> Choice {
+        self.0.ct_gt(&other.0)
+    }
+}
+
+impl ConstantTimeLess for IsvSvn {}
+
 /// CPU security version number (SVN)
 #[repr(transparent)]
-#[derive(Default, Debug, Clone, Hash, PartialEq, Eq)]
+#[derive(Default, Debug, Clone, Hash, PartialEq, Eq, ConstantTimeEq)]
 pub struct CpuSvn(sgx_cpu_svn_t);
 
 impl_newtype_for_bytestruct! {
     CpuSvn, sgx_cpu_svn_t, SGX_CPUSVN_SIZE, svn;
 }
 
+impl ConstantTimeGreater for CpuSvn {
+    fn ct_gt(&self, other: &Self) -> Choice {
+        if !self.0.svn.len().eq(&other.0.svn.len()) {
+            return Choice::from(0);
+        }
```

Comment:
:wrench: This check isn't needed since the array is defined with a size inside of a struct
```suggestion
```